spec/controllers/v3/tags_controller_spec.rb

```diff
@@ -0,0 +1,66 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe V3::TagsController do
+  subject { create :repository_compound }
+  let!(:user) { subject.owner }
+
+  context 'successful' do
+    describe 'action: index' do
+      before do
+        get :index, params: {organizational_unit_id: user.to_param,
+                             repository_id:
+                             subject.to_param.sub(%r{\A[^/]*/}, '')}
+      end
+      it { expect(response).to have_http_status(:ok) }
+      it { |example| expect([example, response]).to comply_with_rest_api }
+      it 'finds the repository' do
+        expect(response_data['repository']).not_to be(nil)
+      end
+    end
+
+    describe 'action: show' do
+      let!(:tag_name) { '1.0' }
+      before do
+        subject.git.create_tag(tag_name, subject.git.default_branch)
+        get :show, params: {organizational_unit_id: user.to_param,
+                            repository_id:
+                            subject.to_param.sub(%r{\A[^/]*/}, ''),
+                            name: tag_name}
+      end
+      it { expect(response).to have_http_status(:ok) }
+      it { |example| expect([example, response]).to comply_with_rest_api }
+      it 'finds the tag' do
+        expect(response_data['repository']['tag']).not_to be(nil)
+      end
```

Comment:
You could replace this a test that checks if the correct tag has been returned:
```ruby
it 'finds the correct tag' do
  expect(response_data['repository']['tag']['name']).to eq(tag_name)
end
```
The not-nil property is checked implicitly with this.